packages/asset-service/src/generateAssetData/osmosis/index.ts

```diff
@@ -17,33 +18,72 @@ type OsmoAsset = {
   name: string
   display: string
   symbol: string
+  traces: {
+    type: string
+    counterparty: {
+      chain_name: string
+      bse_denom: string
+      channel_id: string
+    }
+    chain: {
+      channel_id: string
+    }
+  }[]
   logo_URIs: {
     png?: string
     svg?: string
   }
   coingecko_id: string
-  pools?: {
-    [key: string]: number
-  }
+  keywords: string[]
 }
 
 type OsmosisAssetList = {
-  chain_id: string
+  chain_name: string
   assets: OsmoAsset[]
 }
 
 export const getAssets = async (): Promise<Asset[]> => {
+  /** Helper function to get latest commit hash for default branch of given repo.
+   * This is used to construct githack URLs */
+  const getCurrentHash = memoize(async (repository: string): Promise<string> => {
+    /* Trim any trailing slashes */
+    while (repository.charAt(repository.length - 1) === '/') {
+      repository = repository.slice(0, -1)
+    }
+
+    const {
+      data: { default_branch: defaultBranch },
+    } = await axios.get(`https://api.github.com/repos/${repository}`)
+
+    const {
+      data: {
+        object: { sha: currentHash },
+      },
+    } = await axios.get(
+      `https://api.github.com/repos/${repository}/git/refs/heads/${defaultBranch}`,
+    )
+
+    return currentHash
+  })
+
+  /* Fetch asset list */
+  const currentAssetListsHash = await getCurrentHash('osmosis-labs/assetlists')
+  console.info('currentAssetListsHash', currentAssetListsHash)
   const { data: assetData } = await axios.get<OsmosisAssetList>(
-    'https://raw.githubusercontent.com/osmosis-labs/assetlists/main/osmosis-1/osmosis-1.assetlist.json',
+    `https://raw.githack.com/osmosis-labs/assetlists/${currentAssetListsHash}/osmosis-1/osmosis-1.assetlist.json`,
   )
 
+  if (!assetData) throw new Error('Could not get Osmosis asset data!')
+  console.info('assetData', assetData)
+
   /* Osmosis pool IDs are guaranteed to be unique integers, so we use a set to keep track of 
     which pools we've already seen. A lookup is necessary because the Osmosis asset list 
     contains duplicate entries for each pool, eg. ATOM/OSMO == OSMO/ATOM. 
   */
   const lpAssetsAdded = new Set()
 
-  return assetData.assets.reduce<Asset[]>((acc, current) => {
+  return await assetData.assets.reduce<Promise<Asset[]>>(async (accPrevious, current) => {
```

Comment:
`return await somePromise` is effectively `return somePromise`, since you get a promise out of `getAssets` either way
```suggestion
  return assetData.assets.reduce<Promise<Asset[]>>(async (accPrevious, current) => {
```